app/src/main/java/com/geeksville/mesh/service/MeshService.kt

```diff
@@ -1020,11 +1020,20 @@ class MeshService : Service(), Logging {
                 it.snr = packet.rxSnr
                 it.rssi = packet.rxRssi
             }
-
+            //generate our own hopsAway, comparing hopStart to hopLimit.
+            when {
```

Comment:
setting `hopsAway` needs to move inside the `updateNodeInfo()` block [above](https://github.com/meshtastic/Meshtastic-Android/blob/b69466f669ce360577267e9adc41a970abae0421/app/src/main/java/com/geeksville/mesh/service/MeshService.kt#L1014). extra calls will trigger unnecessary node change broadcasts for every packet received.

this should match how `hopsAway` is calculated in the firmware.

Reply:
I've now moved `hopsAway `in to the `updateNodeInfo()` block as required.

Regarding `hopsAway `calculation, this essentially does follow the Firmware implementation, it checks if `hopStart `is not 0, then `hopStart `- `hopLimit`.

The one thing it does extra from the firmware is it checks if `hopStart` is less than `hopLimit`, if it is, set `hopsAway `to -1. It then uses this to append `(!)` to the RSSI data, to signify that it's probably from a device with old Firmware. 

The firmware version of this implementation does not differentiate between a packet which is received direct, and a packet from a node with old firmware (both return 0). This can be confusing to the user, and may make them think all nodes are local to them.

Is this OK? If not I have no problem removing it.

Reply:
keep it simple and align with the code in the firmware. `hopsAway` from the radio (removed in [`b69466f`](https://github.com/meshtastic/Meshtastic-Android/pull/966/commits/b69466f669ce360577267e9adc41a970abae0421#diff-3a6bc031c4bd5b2872f201d2b4b98ed404f169f548d64821749bd6144efaf854L1322)) must be used and any extra handling must be compatible.

keep in mind all this added complexity will be unnecessary in a few weeks once everyone upgrades to 2.3.0 or later. if there is room for improvement here or on the firmware implementation, address them in separate PRs.

Reply:
no problem, I've now removed the extras and `hopsAway` now follows the firmware implementation.

Reply:
thanks